### PR TITLE
update go to 1.22.5 to fix vulnerable package net/http

### PR DESCRIPTION
**Pull request by pjuarezd**
```
Vulnerability #1: GO-2024-2963
    Denial of service due to improper 100-continue handling in net/http
  More info: https://pkg.go.dev/vuln/GO-2024-2963
  Standard library
    Found in: net/http@go1.22.4
    Fixed in: net/http@go1.22.5
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,6 @@
 module github.com/minio/operator
 
-go 1.22.0
-
-toolchain go1.22.4
+go 1.22.5
```
Comment (harshavardhana) [inline go.mod:3]:
```suggestion
go 1.22
```

File: sidecar/go.mod [hunk 2]
```diff
@@ -1,8 +1,6 @@
 module github.com/minio/operator/sidecar
 
-go 1.22.0
-
-toolchain go1.22.4
+go 1.22.5
```
Comment (harshavardhana) [inline sidecar/go.mod:3]:
```suggestion
go 1.22
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (pjuarezd) [reply go.mod]: `go mod tidy`  is adding automatically the `.0` at the end, or maybe is just my IDE, but this seems fine to me, we just need to make sure is not added back again

Comment (pjuarezd) [reply go.mod]: nope, not  building @harshavardhana

**Comment by pjuarezd** [comment]
FYI build is not passing with only `1.22` @harshavardhana

**Comment by harshavardhana** [comment]
will take a look @pjuarezd

**Comment by harshavardhana** [comment]
Yeah its because of stupid dependency update

```
go: k8s.io/api@v0.30.2 requires go@1.22.0, but 1.22 is requested
```

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**